src/NameFieldSet.tsx

```diff
@@ -114,12 +113,13 @@ export class NameFieldSet extends React.Component<INameFieldSetProps, INameField
     return (index !== -1 && index <= 1);
   }
 
-  private removeBlankNamesFromNameArray = (newNames: string[]) => {
-    for (let index = 0; index <= newNames.length; index++) {
-      if (this.isEmptyOrSpaces(newNames[index])) {
-        newNames.splice(index, 1);
+  private removeBlankNamesFromNameArray = (names: string[]) => {
+    for (let index = 0; index <= names.length; index++) {
+      if (this.isEmptyOrSpaces(names[index])) {
+        names.splice(index, 1);
       }
     }
```

Comment:
This for loop modifies the parameter object, which is not ideal.

You can achieve the same thing by using a more functional approach:

return names.filter((name) => !this.isEmptyOrSpaces(name));